src/SFA.DAS.AAN.Hub.Data/Repositories/JobAuditRepository.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AAN.Hub.Data.Entities;
+using SFA.DAS.AAN.Hub.Data.Interfaces;
+using System;
+
+namespace SFA.DAS.AAN.Hub.Data.Repositories
+{
+    public class JobAuditRepository : IJobAuditRepository
+    {
+        private readonly ILogger<JobAuditRepository> _logger;
+        private readonly IAanDataContext _context;
+
+        public JobAuditRepository(ILogger<JobAuditRepository> logger, IAanDataContext context) 
+        { 
+            _logger = logger;
+            _context = context;
+        }
+
+        public async Task<JobAudit?> GetMostRecentJobAudit(CancellationToken cancellationToken)
+        {
+            try
+            {
+                return await _context.JobAudits
+                                .AsNoTracking()
+                                .OrderByDescending(a => a.StartTime)
+                                .FirstOrDefaultAsync(cancellationToken);
+            }
+            catch(Exception _exception)
+            {
+                _logger.LogError(_exception, "Unable to get most recent successful audit record.");
+                return null;
+            }
+        }
+
+        public async Task RecordAudit(JobAudit jobAudit, CancellationToken cancellationToken)
+        {
+            try
+            {
+                _context.JobAudits.Add(jobAudit);
+                await _context.SaveChangesAsync(cancellationToken);
```

Comment:
Save changes should not be added to the repository as it will be non-transactional commit. Remove save changes from the repository and move it into service. So all the changes are save under one transaction. FYI the dbcontext is scoped in DI so an intrinsic transaction is created for you.